### PR TITLE
Feature/filestorage sharetype getextraspecs

### PR DESCRIPTION
**Pull request by ehdou**
Add support for Share Type Get Extra Specs
For #140

### CODE REVIEW

**Comment by ehdou** [comment]
[Get extra specs function](https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/share_types_extra_specs.py#L33) for share type

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review. It's based on [PR#144](https://github.com/gophercloud/gophercloud/pull/144) and adds support for share type Get Extra Specs

**Comment by ehdou** [comment]
@jrperritt This PR is ready for review

File: openstack/sharedfilesystems/v2/sharetypes/results.go [hunk 1]
```diff
@@ -68,3 +68,28 @@ func ExtractShareTypes(r pagination.Page) ([]ShareType, error) {
 type GetDefaultResult struct {
 	commonResult
 }
+
+// ExtraSpecs contains all the information associated with extra specifications
+// for an Openstack ShareType.
+type ExtraSpecs struct {
```
Comment (jrperritt) [inline openstack/sharedfilesystems/v2/sharetypes/results.go]: If `ExtraSpecs` is just a map with [varying] key-value pairs, we don't need to wrap it in a `struct`:
```go
type ExtraSpecs map[string]interface{}
```

**Comment by ehdou** [comment]
@jrperritt This PR is ready for re-review

**Comment by jrperritt** [comment]
+2

**Pull request merged**